Lib/test/test_ctypes/test_refcounts.py

```diff
@@ -97,5 +97,31 @@ def func(a, b):
         f(1, 2)
         self.assertEqual(sys.getrefcount(ctypes.c_int), a)
 
+    @support.refcount_test
+    def test_callback_py_object_none_return(self):
+        """Test that returning ``None`` from a ``py_object`` callback
+        does not affect ``None``'s refcount (bpo-36880)."""
+
+        import sys
+
+        for FUNCTYPE in (ctypes.CFUNCTYPE, ctypes.PYFUNCTYPE):
+            with self.subTest(FUNCTYPE=FUNCTYPE):
+                proto = FUNCTYPE(ctypes.py_object)
+                @proto
+                def func():
+                    return None
+
+                # Check that calling func does not affect None's refcount.
+                none_refcount = sys.getrefcount(None)
+                # Because None's refcount can also change for other reasons,
+                # we call func in a loop to ensure that any effects on None's
+                # refcount are clearly visible.
+                for _ in range(10000):
+                    func()
+                # Allow for small variations in None's refcount from other
+                # sources.
+                self.assertAlmostEqual(
+                    sys.getrefcount(None), none_refcount, delta=50)
```

Comment:
Yeah, in general we don't do these kind of tests. We have an automatic refleak detection mechanism but it will only work if the path that leaks references is exercised in the test suite. This means that you can add a test that ensures that the code you are adding to `callbacks.c` is actually executed but your test doesn't need to try to detect the refleak, that will be done automatically. I suppose the reason this was never detected is that we don't have a test that actually runs that.

Reply:
Ah okay. I was going by what the other tests in `test_refcounts` are doing. Should I adjust those as well to not check `sys.getrefcount` manually?

Reply:
> Should I adjust those as well to not check `sys.getrefcount` manually?

Not as part of this issue (one issue per problem - only changing what's required for the problem at hadn). I haven't checked those and maybe these are checked more specific things.

Reply:
Thanks, I didn't know that just `-m test` works too (I used `-m test.regrtest`). Yes, I checked - with `-R` it now indeed detects the original problem.